src/reducers/sources.js

```diff
@@ -71,15 +71,15 @@ function update(
       }
 
       location = {
-        line: action.line,
+        ...action.location,
         url: action.source.url
       };
 
       prefs.pendingSelectedLocation = location;
       return state
         .set("selectedLocation", {
           sourceId: action.source.id,
-          line: action.line
+          location: action.location
```

Comment:
i think this is the source of the redundancy... 
we should probably just be doing

```js
.set("selectedLocation", action.location)
```

Reply:
I'm thinking
```js 
.set("selectedLocation", {
   sourceId: action.source.id,
   ...action.location
})
```
as we need the `sourceId` property